lib/ansible/modules/messaging/kafka_topics.py

```diff
@@ -0,0 +1,338 @@
+'''
+
+EXAMPLES = '''
+name: 'list all topic'
+kafka_topics:
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+
+name: 'list specific topic'
+kafka_topics:
+  topic: 'foobar'
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+
+name: 'Create topic foobar'
+kafka_topics:
+  action: 'create'
+  topic: 'foobar'
+  config:
+    cleanup.policy: 'delete'
+    compression.type: 'gzip'
+  partitions: 2
+  replication_factor: 1
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+
+# If you have JAAS authentification,
+# use environment variable
+name: 'Create topic foobar'
+environment:
+  KAFKA_OPTS: '-Djava.security.auth.login.config=/etc/kafka/jaas.conf'
+kafka_topics:
+  action: 'create'
+  topic: 'foobar'
+  config:
+    cleanup.policy: 'delete'
+    compression.type: 'gzip'
+  partitions: 2
+  replication_factor: 1
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+
+name: 'describe specific topic foobar'
+kafka_topics:
+  action: 'describe'
+  topic: 'foobar'
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+
+name: 'alter partitions for topic foobar'
+kafka_topics:
+  action: 'alter'
+  topic: 'foobar'
+  partitions: 3
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+
+name: 'delete topic foobar'
+kafka_topics:
+  action: 'delete'
+  topic: 'foobar'
+  zookeeper:
+    - zookeeper1.foo.bar:2181
+    - zookeeper2.foo.bar:2181
+'''
+
+RETURN = '''
+stdout:
+  description: Output from stdout of kafka-topics command after execution of given command.
+  returned: success or changed
+  type: string
+  sample: "Created topic raclette"
+
+stderr:
+  description: kafka-topics command execution stderr.
+  returned: failed
+  type: string
+  sample: "[2017-07-11 14:46:04,092] ERROR org.apache.kafka.common.errors.InvalidTopicException: topic name [foobar] is illegal"
+
+failed:
+  description: kafka-topics command execution return bool.
+  returned: always
+  type: bool
+  sample: False
+'''
+
+import re
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+class KafkaError(Exception):
+    pass
+
+
+class KafkaTopics(object):
+
+    def __init__(self, module):
+
+        if module.params['config']:
+            self.config = ''.join(' --config %s=%r' % (key, value) for (key, value)
+                                  in module.params['config'].items())
+        else:
+            self.config = ''
+
+        self.partitions = module.params['partitions']
+        self.replication_factor = module.params['replication_factor']
+        self.topic = module.params['topic']
+        self.zookeeper = ','.join(module.params['zookeeper'])
+        self.action = module.params['action']
+
+        self.executable = module.params['executable'] or module.get_bin_path('kafka-topics', True)
```

Comment:
There is a default value for `executable` module parameter: then `get_bin_path` will never be called.